core/network/src/consensus_gossip.rs

```diff
@@ -61,12 +70,20 @@ pub enum ValidationResult<H> {
 pub trait Validator<H> {
 	/// Validate consensus message.
 	fn validate(&self, data: &[u8]) -> ValidationResult<H>;
+
+	/// Produce a closure for validating messages on a given topic.
+	fn message_expired<'a>(&'a self) -> Box<FnMut(H, &[u8]) -> bool + 'a> {
```

Comment:
Why return a closure instead of just calling the method directly like we do for validate?

Reply:
my intuition was that this can get called with a lot of topic messages at once, so the closure lets us hold locks